tests/integration-tests/src/test/java/org/apache/activemq/artemis/tests/integration/http/HttpAuthorityTest.java

```diff
@@ -0,0 +1,111 @@
+import io.netty.channel.nio.NioEventLoopGroup;
+import io.netty.channel.socket.SocketChannel;
+import io.netty.channel.socket.nio.NioServerSocketChannel;
+import io.netty.handler.codec.http.DefaultFullHttpResponse;
+import io.netty.handler.codec.http.FullHttpRequest;
+import io.netty.handler.codec.http.HttpObjectAggregator;
+import io.netty.handler.codec.http.HttpResponseStatus;
+import io.netty.handler.codec.http.HttpServerCodec;
+import io.netty.handler.stream.ChunkedWriteHandler;
+import org.apache.activemq.artemis.api.core.client.ActiveMQClient;
+import org.apache.activemq.artemis.api.core.client.ClientSessionFactory;
+import org.apache.activemq.artemis.api.core.client.ServerLocator;
+import org.apache.activemq.artemis.tests.util.ActiveMQTestBase;
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+public class HttpAuthorityTest extends ActiveMQTestBase {
+
+   @Test
+   public void testHttpAuthority() throws Exception {
+      int port = 61616;
+      CountDownLatch requestTested = new CountDownLatch(1);
+      AtomicBoolean failed = new AtomicBoolean(false);
+      EventLoopGroup bossGroup = new NioEventLoopGroup();
+      EventLoopGroup workerGroup = new NioEventLoopGroup();
+      try {
+         ServerBootstrap bootstrap = new ServerBootstrap();
+         bootstrap.group(bossGroup, workerGroup).channel(NioServerSocketChannel.class).childHandler(new ChannelInitializer<SocketChannel>() {
+            @Override
+            public void initChannel(SocketChannel ch) {
+               ch.pipeline().addLast(new HttpServerCodec());
+               ch.pipeline().addLast(new HttpObjectAggregator(1024 * 1024));
+               ch.pipeline().addLast(new ChunkedWriteHandler());
+               ch.pipeline().addLast(new SimpleChannelInboundHandler<FullHttpRequest>() {
+                  @Override
+                  protected void channelRead0(ChannelHandlerContext ctx, FullHttpRequest req) {
+                     if (failed.get()) {
+                        return;
+                     }
+
+                     try {
+                        failed.set(!new URI(req.uri()).getAuthority().equals(req.headers().get("host")));
+                     } catch (Exception e) {
+                        failed.set(true);
+                     }
+                     requestTested.countDown();
+
+                     DefaultFullHttpResponse response = new DefaultFullHttpResponse(req.protocolVersion(), HttpResponseStatus.OK);
+                     ctx.writeAndFlush(response).addListener(ChannelFuture -> {
+                        if (!ChannelFuture.isSuccess()) {
+                           ChannelFuture.cause().printStackTrace();
+                        }
+                     });
+                  }
+
+                  @Override
+                  public void exceptionCaught(ChannelHandlerContext ctx, Throwable cause) {
```

Comment:
Is the test expected to fail when exceptionCaught is called?

Reply:
No longer relevant.